bitflags.h

```diff
@@ -33,59 +35,136 @@ namespace gcpp {
 	//----------------------------------------------------------------------------
 
 	class bitflags {
+		using unit = unsigned int;
+		static_assert(std::is_unsigned<unit>::value, "unit must be an unsigned integral type.");
+
+		std::unique_ptr<unit[]> bits;
 		const int size;
-		std::vector<byte> bits;
+
+		static constexpr std::size_t bits_per_unit = sizeof(unit) * CHAR_BIT;
```

Comment:
I gave `int` vs. `size_t` a bit of thought, and came down on the side of "it doesn't really matter for correctness / performance." I do agree that `int` would be more consistent with the rest of the design.

The `static constexpr` member needs a definition only if it is ODR-used. The rules for ODR-use are ridiculously complex, but can be reasonably approximated by "it's an ODR-use if you take an address/reference, and not an ODR-use if you immediately perform lvalue-to-rvalue conversion." With a bit of care for how we use the variable, it's possible to avoid defining it.